Mathlib/Algebra/Star/Unitary.lean

```diff
@@ -174,22 +175,67 @@ end Monoid
 
 section Map
 
-variable {F R S : Type*} [Monoid R] [StarMul R] [Monoid S] [StarMul S]
-variable [FunLike F R S] [StarHomClass F R S] [MonoidHomClass F R S] (f : F)
+variable {R S T : Type*} [Monoid R] [StarMul R] [Monoid S] [StarMul S] [Monoid T] [StarMul T]
 
-lemma map_mem {r : R} (hr : r ∈ unitary R) : f r ∈ unitary S := by
+lemma map_mem {F : Type*} [FunLike F R S] [StarHomClass F R S] [MonoidHomClass F R S]
+    (f : F) {r : R} (hr : r ∈ unitary R) : f r ∈ unitary S := by
   rw [unitary.mem_iff] at hr
   simpa [map_star, map_mul] using And.intro congr(f $(hr.1)) congr(f $(hr.2))
 
-/-- The group homomorphism between unitary subgroups of star monoids induced by a star
-homomorphism -/
+/-- The star monoid homomorphism between unitary subgroups induced by a star monoid homomorphism of
+the underlying star monoids. -/
 @[simps]
-def map : unitary R →* unitary S where
+def map (f : R →⋆* S) : unitary R →⋆* unitary S where
   toFun := Subtype.map f (fun _ ↦ map_mem f)
   map_one' := Subtype.ext <| map_one f
   map_mul' _ _ := Subtype.ext <| map_mul f _ _
+  map_star' _ := Subtype.ext <| map_star f _
 
-lemma toUnits_comp_map : toUnits.comp (map f) = (Units.map f).comp toUnits := by ext; rfl
+@[simp]
+lemma coe_map (f : R →⋆* S) (x : unitary R) : (map f x : S) = f x := rfl
```

Comment:
Does the statement change without the manual coercion `(... : S)`?

Reply:
no, I've removed the type ascription